sympy/polys/domainmatrix.py

```diff
@@ -610,6 +614,22 @@ def rref(self):
         rref_ddm, pivots = self.rep.rref()
         return self.from_ddm(rref_ddm), tuple(pivots)
 
+    def nullspace(M):
+        rows, cols = M.shape
+        domain = M.domain
+        rref, pivots = M.rep.rref()
```

Comment:
The fact that this method needs to mess around with the internal of the underlying DDM means that it should be implemented at the DDM level. There should be a method `DDM.nullspace()` that returns `DDM` instances. The code below wouldn't work e.g. for SDM.